mopro-ffi/src/circom/mod.rs

```diff
@@ -67,26 +62,14 @@ pub fn generate_circom_proof_wtns(
     // check the prime in the header
     // println!("{} {} {}", header.q, header.n8q, ark_bls12_381::Fq::MODULUS);
     if header_reader.r == BigUint::from(ark_bn254::Fr::MODULUS) {
-        let mut binfile = BinFile::<_, Bn254>::new(&mut reader)
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let proving_key = binfile
-            .proving_key()
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let matrices = binfile
-            .matrices()
+        let (proving_key, matrices) = read_zkey::<_, Bn254>(&mut reader)
             .map_err(|e| MoproError::CircomError(e.to_string()))?;
         let full_assignment = witness_thread
             .join()
             .map_err(|_e| MoproError::CircomError("Failed to generate witness".to_string()))?;
         return prove(proving_key, matrices, full_assignment);
     } else if header_reader.r == BigUint::from(ark_bls12_381::Fr::MODULUS) {
-        let mut binfile = BinFile::<_, Bls12_381>::new(&mut reader)
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let proving_key = binfile
-            .proving_key()
-            .map_err(|e| MoproError::CircomError(e.to_string()))?;
-        let matrices = binfile
-            .matrices()
+        let (proving_key, matrices) = read_zkey::<_, Bls12_381>(&mut reader)
```

Comment:
I think we should keep these functions separate. `read_zkey` loads both the proving key and the matrices. In the verifier we only want to load the proving key, not the matrices. That's why i originally separated it to two functions.